tsl/src/fdw/deparse.c

```diff
@@ -867,6 +867,60 @@ deparseDistinctClause(StringInfo buf, deparse_expr_cxt *context)
 			return;
 	}
 
+	if (query->hasDistinctOn)
+	{
+		/*
+		 * Pushing down DISTINCT ON is more complex than plain DISTINCT.
+		 * The DISTINCT ON columns must be a prefix of the ORDER BY columns.
+		 * Without this, the DISTINCT ON would return an unpredictable row
+		 * each time. There is a diagnostic for the case where the ORDER BY
+		 * clause doesn't match the DISTINCT ON clause, so in this case we
+		 * would get an error on the data node. There is no diagnostic for
+		 * the case where the ORDER BY is absent, so in this case we would
+		 * get a wrong result.
+		 * The remote ORDER BY clause is created from the pathkeys of the
+		 * corresponding relation. If the DISTINCT ON columns are not a prefix
+		 * of these pathkeys, we cannot push it down.
+		 */
+		int next_pathkey_index = 0;
+		foreach (l, query->distinctClause)
+		{
+			SortGroupClause *sgc = lfirst_node(SortGroupClause, l);
+			TargetEntry *tle = get_sortgroupclause_tle(sgc, query->targetList);
+
+			if (next_pathkey_index >= list_length(pathkeys))
+			{
+				return;
+			}
```

Comment:
```suggestion
			if (next_pathkey_index >= list_length(pathkeys))
				return;
```

Reply:
I'm going to fix this and I know that this is postgres style we have to follow, but: this is widely considered a bad practice (maybe you remember the Apple's `goto fail;`). I personally tend to have a lot of problems with this while editing the postgres code, because I forget to add the braces, in the same `goto fail` style. Unfortunately, clang-format can't fix this, and clang-tidy can only add braces, not remove.